packages/grid/_modules_/grid/GridComponent.tsx

```diff
@@ -55,6 +55,24 @@ import { visibleGridRowCountSelector } from './hooks/features/filter/gridFilterS
 
 export const GridComponent = React.forwardRef<HTMLDivElement, GridComponentProps>(
   function GridComponent(props, ref) {
+    const {
+      apiRef: apiRefProp,
+      columns,
+      rows,
+      getRowId,
+      components: componentsProp,
+      componentsProps,
+      state,
+      loading,
+      className,
+      licenseStatus,
+      nonce,
+      autoHeight,
+    } = props;
+
+    // freeze rows for immutability
+    Object.freeze(rows);
```

Comment:
Whats the reason? Why? 🤔

Reply:
The freeze is meant to help developers use the API correctly, to have a fast feedback loop when they do something wrong. In production, it might lead to a crash in edge cases. What's the use case for having it in production?

Reply:
Like what if they ignore or miss that in dev mode but still keep mutating in prod build.

Reply:
> what if

We take the risk to break the UX for a change that was done for DX motivations.